ios/Classes/Embeds/AppcuesFrameViewFactory.swift

```diff
@@ -0,0 +1,195 @@
+import Flutter
+import UIKit
+import AppcuesKit
+
+class AppcuesFrameViewFactory: NSObject, FlutterPlatformViewFactory {
+    private let messenger: FlutterBinaryMessenger
+    private let plugin: SwiftAppcuesFlutterPlugin
+
+    init(plugin: SwiftAppcuesFlutterPlugin, messenger: FlutterBinaryMessenger) {
+        self.plugin = plugin
+        self.messenger = messenger
+        super.init()
+    }
+
+    func create(
+        withFrame frame: CGRect,
+        viewIdentifier viewId: Int64,
+        arguments args: Any?
+    ) -> FlutterPlatformView {
+        return AppcuesPlatformView(
+            plugin: plugin,
+            frame: frame,
+            viewIdentifier: viewId,
+            arguments: args,
+            binaryMessenger: messenger)
+    }
+
+    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
+    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
+          return FlutterStandardMessageCodec.sharedInstance()
+    }
+}
+
+class AppcuesPlatformView: NSObject, FlutterPlatformView {
+    private var _view: UIView
+
+    init(
+        plugin: SwiftAppcuesFlutterPlugin,
+        frame: CGRect,
+        viewIdentifier viewId: Int64,
+        arguments args: Any?,
+        binaryMessenger messenger: FlutterBinaryMessenger?
+    ) {
+        let args = args as? [String: Any]
+
+        _view = WrapperView(
+            plugin: plugin,
+            binaryMessenger: messenger,
+            frameID: args?["frameId"] as? String,
+            viewIdentifier: viewId
+        )
+        super.init()
+    }
+
+    func view() -> UIView {
+        return _view
+    }
+}
+
+
+class WrapperView: UIView, FlutterStreamHandler {
+
+
+    private let plugin: SwiftAppcuesFlutterPlugin
+    private var eventChannel: FlutterEventChannel?
+    private var eventSink: FlutterEventSink?
+
+    var frameID: String? = nil
+
+    private weak var frameViewController: AppcuesFrameVC?
+
+    init(
+        plugin: SwiftAppcuesFlutterPlugin,
+        binaryMessenger messenger: FlutterBinaryMessenger?,
+        frameID: String?,
+        viewIdentifier viewId: Int64
+    ) {
+        self.plugin = plugin
+        self.frameID = frameID
+
+        if let messenger = messenger {
+            eventChannel = FlutterEventChannel(name: "com.appcues.flutter/frame/\(viewId)", binaryMessenger: messenger)
+        }
+
+        super.init(frame: .zero)
+
+        eventChannel?.setStreamHandler(self)
```

Comment:
this is what opens the `EventChannel` and ends up creating the `eventSync` value used later in this view.